Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,396 @@
+
+_Example: scopes of automatic "purpose\_1" and "purpose\_2" have been granted, and  "authorization\_id" is provided in the response._
+
+```http
+POST /token
+Content-Type: application/x-www-form-urlencoded
+Authorization: Basic czZCaGRSa3F0MzpnWDFmQmF0M2JW
+
+grant_type=authorization_code&code=SplxlOB&redirect_uri= https%3A%2F%2Fclient%2Eexample%2Ecom%2Fcb
+
+HTTP/1.1 200 OK
+Content-Type: application/json;charset=UTF-8
+
+{
+  "access_token":"2YotnFZFEjr1zCsicMWpAA",
+  "token_type":" Bearer",
+  "authorization_id":"AAAorm2v3v4b3n4otnXdFtGhUhDf",
+  "expires_in":3600,
+  "scope":"pi_scope_1_of_purpose_1 pi_scope_2_of_purpose_1 pi_scope_1_of_purpose_2",
+  "purpose":"purpose_1 purpose_2"
+}
+```
+
+## Obtaining an access token for non-automatic purposes
+
+The OpenID Connect specification does not define how to get access tokens for scopes that require consent management. It is basically a recommendation described in the [Open ID Connect Specification](https://openid.net/specs/openid-connect-core-1_0.html#Consent) about how the Auth server can capture the consent of the requested scopes, that it MAY be done through an interactive dialogue with the end-user.
+
+There is also no specific solution defined for a scenario where the Auth server cannot capture the consent and it must be delegated to a third party.
+
+For the Operator NaaS platform, it is needed to define a delegated authorization procedure for an application backend to obtain access tokens for purposes which require explicit user consent. This procedure MUST rely upon existing previous authorization session of the user (i.e. using the `authorization id` result of a previous login authorization flow where user identity was validated against the corresponding IdP). This procedure does not require further user interaction in between, once the user consent is collected. Consent gathering from users is a one-off independent process that must be completed before requesting an access token to access the required Personal Information. The **Consent API** is used to register user consent.
+
+The proposal is to use the [Assertion Framework for OAuth 2.0 Client Authentication and Authorization Grants (RFC 7521)](https://datatracker.ietf.org/doc/html/rfc7521) specification using an assertion of type JWT as described in [JSON Web Token (JWT) Profile for OAuth 2.0 Client Authentication and Authorization Grants (RFC 7523)](https://datatracker.ietf.org/doc/html/rfc7523), which uses the "**urn:ietf:params:oauth:grant-type:jwt-bearer**" grant type. The figure below illustrates the workflow.
+
+![Delegated authorization jwt-bearer flow](./images/userconsent-1.png)
```

Comment:
In userconsent-1.png, it would be good to keep the participants consistent to userconsent-2.png and specify that AuthServer is a part of NaaS Operator